### ISSUE TITLE
az image create -- Limited command options as compared to Azure Portal

### ISSUE THREAD

**Issue by SimSama** [Azure/azure-cli#14584]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**

Perhaps a maturity issue as opposed to bug. The documentation for az image create does not list a clear way to specify storage-sku for each data disk attached to an image. 

Also encryption options visible in Image create UI, is not listed in configuration options in Azure CLI. 

Sample command: -- this is a long multi-line command in windows cmd. It works, but does not allow specifying encryption at rest, or per data disk storage type, as seen in the GUI.

az image create --name image1 -g resource_group01 --os-type Linux --location eastus --hyper-v-generation V1 ^
 --zone-resilient true --storage-sku Premium_LRS ^
 --source https://storage1.blob.core.windows.net/container1/root-disk.vhd ^
 --data-disk-sources https://storage1.blob.core.windows.net/container1/second-disk.vhd https://storagestratozen.blob.core.windows.net/container1/third-disk.vhd ^
 --os-disk-caching ReadWrite ^
 --data-disk-caching ReadWrite ^

//Relevant documentation
https://docs.microsoft.com/en-us/cli/azure/image?view=azure-cli-latest#az-image-create

In Summary:

1) How do I specify encryption for the image, as seen in the UI?
2) How do I specify storage type (premium etc) per data disk if multiple?
3) How do I specify disk read/write caching per data disk added? -- I mean usually I'll do read/write for everything, but why does the option not exist?

Is there a simple way to do this via azure cli, or do I have to use some way to input json (no example documentation).
Is the only option an ARM template?

**Comment by arrownj** [comment]
Hi @qwordy , could you please help with this issue ? Thanks

**Comment by qwordy** [comment]
Reasonable ask. Added to S178. It is a known issue. Azure CLI doesn't guarantee that it supports all features defined in specification. Neither does Portal. Azure CLI and Portal are designed separately by different people. Sometimes they have differences.

**Comment by z0n3z3r0** [comment]
Thanks! I look forward to seeing it in upcoming release.

**Comment by yungezz** [comment]
hi @qwordy could you pls update status of the issue? thanks.

**Comment by qwordy** [comment]
This feature request is in the queue and does not have a definite timeline now.

**Comment by zhoxing-ms** [comment]
> How do I specify encryption for the image, as seen in the UI?
   How do I specify storage type (premium etc) per data disk if multiple?
   How do I specify disk read/write caching per data disk added? -- I mean usually I'll do read/write for everything, but why does the option not exist?

The REST documentation [images/create-or-update](https://docs.microsoft.com/en-us/rest/api/compute/images/create-or-update) doesn't seem to show support for these features, which requires service team to confirm whether support is required

**Comment by olayemio** [comment]
@SimSama user images do not support encryption and there are no plans to support it. We encourage our customers to move over to the [Shared Image Gallery](https://docs.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries). You can import your user image into the Shared Image Gallery by following these [instructions](https://docs.microsoft.com/en-us/azure/virtual-machines/image-version-managed-image-cli).

**Comment by SimSama** [comment]
> @SimSama user images do not support encryption and there are no plans to support it. We encourage our customers to move over to the [Shared Image Gallery](https://docs.microsoft.com/en-us/azure/virtual-machines/shared-image-galleries). You can import your user image into the Shared Image Gallery by following these [instructions](https://docs.microsoft.com/en-us/azure/virtual-machines/image-version-managed-image-cli).

I think you are getting caught up in word semantics. Disk encryption for custom linux images seems to be supported in the console. If you are trying  to say disk encryption is not supported for any non microsoft supplied image, then we'd move our customers off Azure.

I only ask for features supported in the console as of today, and 6+ months ago be replicated in REST / cmd. These things are already implemented features as of this moment, just not in all configuration methods. 

These are all options I had available to me in the GUI, but no equivalent option in the command line, so to say they don't work or not supported outright just seems wrong.

If you don't want to fix it, the discussion is over it seems. More revenue for AWS.

**Comment by olayemio** [comment]
I misspoke. The API for user images does support encryption for data disks. 
@zhoxing-ms examples may be missing but the [Images API supports Disk Encryption Set](https://docs.microsoft.com/en-us/rest/api/compute/images/create-or-update#imagedatadisk). 
@qwordy is your team still working on supporting this?

**Comment by zhoxing-ms** [comment]
> How do I specify encryption for the image, as seen in the UI?
How do I specify storage type (premium etc) per data disk if multiple?
How do I specify disk read/write caching per data disk added? -- I mean usually I'll do read/write for everything, but why does the option not exist?

@olayemio Hi, our team is working on CLI. May I ask does REST API support these features now?
I have not found the relevant documentation at this time. If service supports them, how should CLI pass in parameters to REST? It would be better if you could give us some examples~

**Comment by olayemio** [comment]
@zhoxing-ms Yes, all these features are supported in the [REST API](https://docs.microsoft.com/en-us/rest/api/compute/images/create-or-update#imagedatadisk). Customers are only able to specify all three settings if they use disks, snapshots or blobs to create the image. For customers that use VMs to create their image, the settings have to be applied to the disks on the VM before the image capture process.

Here is a sample that has uses encryption, specifies storage type, and the caching.

```
{
  "location": "East US",
  "properties": {
    "storageProfile": {
      "osDisk": {
        "osType": "Linux",
        "snapshot": {
          "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/disks/{osDiskName}"
        },
        "diskEncryptionSet": {
          "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/diskEncryptionSets/{desName}"
        },
        "caching": "None",
        "osState": "Generalized"
      },
      "dataDisks": [
        {
          "lun": 0,
          "managedDisk": {
            "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/disks/{diskName0}"
          },
          "diskEncryptionSet": {
            "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/diskEncryptionSets/{desName}"
          },
          "storageAccountType": "Premium_LRS",
          "caching": "ReadOnly"
        },
        {
          "lun": 1,
          "managedDisk": {
            "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/disks/{diskName1}"
          },
          "diskEncryptionSet": {
            "id": "/subscriptions/[subid}/resourceGroups/{rg}/providers/Microsoft.Compute/diskEncryptionSets/{desName}"
          },
          "storageAccountType": "Standard_LRS",
          "caching": "ReadWrite"
        }
      ],
      "zoneResilient": false
    }
  }
}
```

**Comment by zhoxing-ms** [comment]
@olayemio  OK, I see. Thank you for the example. We will arrange development for this requirement~
